### ISSUE TITLE
Aliasing languages in scoped context for a language map?

### ISSUE THREAD

**Issue by azaroth42** [w3c/json-ld-syntax#158]
It's possible to alias `@none` to `none` in a scoped context of a language map, but should it also be possible to alias other language keys?  For example, if `en-gb` was aliased to `en_gb`, then it would be usable as a property in javascript.

**Comment by gkellogg** [comment]
Terms expand to IRIs, so it's not possible to define an alias for "en-gb" which is anything other than an IRI. The exception is for aliases to keywords, so `@none` works.

The algorithm does try to expand keys for all maps, including language maps, which is why `@none` aliasing works, but we're limited by how terms can be defined.

**Comment by azaroth42** [comment]
Yup, agreed that we can't do it now (because I tried, and got the error :) ) but is it valuable enough to add it?

**Comment by pchampin** [comment]
I can sympathize with the argument that "en-gb" is not usable as a JS property, but I think that a general aliasing mechanism would add too much complexity.

What about creating a special case for language maps, replacing all underscores in the keys by dashes (if I'm correct, BCP 47 tags can not contain underscores)?

This does add complexity to the overall spec (but much smaller than a general alias mechanism), and it does create a backward incompatibility (but only for cases which are somehow invalid anyway). Also, we this would create round-tripping problems...

Is it really worth it?

**Comment by pchampin** [comment]
Note also that it is always possible (although not as convenient) to use an index map:
```json
{
  "@context": {
    "label": { "@id": "http://example.org/label", "@container": "@index" }
  },
  "label": {
    "en_gb": {
      "@value": "hello",
      "@language": "en-gb"
    }
  }
}
```

**Comment by iherman** [comment]
This issue was discussed in [a meeting](https://www.w3.org/2018/json-ld-wg/Meetings/Minutes/2019/2019-04-19-json-ld#section4-1).

- `RESOLVED:  Close syntax #158 won’t fix, too complicated for the value gained`
<details><summary><i class="differentiate">View the transcript</i></summary>
language aliasing<br/>
Rob Sanderson: link: <a href="https://github.com/w3c/json-ld-syntax/issues/158">https://github.com/w3c/json-ld-syntax/issues/158</a><br/>
Rob Sanderson: Unless there are other people that are interested in this particular topic, I’m happy to close it.<br/>
… We can alias @none to just none to get rid of @.<br/>
Gregg Kellogg: Aliasing lets us alias IRIs and keywords, but that’s it.<br/>
Gregg Kellogg: Because @none is a keyword, we can alias it but we can’t alias arbitrary string values.<br/>
Rob Sanderson: Yes, the thought was if we wanted to let some way of aliasing more things then en-us could be aliased to en_us for example. But as Gregg said that means introducing a new feature to do this. Unless there’s interest, I’m ready to close.<br/>
Proposed resolution: Close syntax #158 won’t fix, too complicated for the value gained (Rob Sanderson)<br/>
Rob Sanderson: +1<br/>
Gregg Kellogg: +1<br/>
Ruben Taelman: +1<br/>
Ivan Herman: +1<br/>
Dave Longley: +1<br/>
Resolution #2: Close syntax #158 won’t fix, too complicated for the value gained<br/>
David I. Lehn: +1<br/>
</details>
